lib/orbit/certification.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Orbit.Certification do
+  use Ecto.Schema
+  import Ecto.Changeset
+  import Ecto.Query
+
+  alias Orbit.Certification
+  alias Orbit.RailLine
+  alias Orbit.Repo
+
+  # Over in Glides "rail" is referred to as "operator", but it's the same thing.
+  @type certification_type :: :right_of_way | :rail
+
+  @type t :: %__MODULE__{
+          badge: String.t(),
+          type: certification_type(),
+          rail_line: RailLine.t(),
+          expires: Date.t()
+        }
+  schema "certifications" do
+    field(:badge, :string)
+    field(:type, Ecto.Enum, values: [right_of_way: 0, rail: 1])
+    field(:rail_line, RailLine)
+    field(:expires, :date)
+    timestamps()
+  end
+
+  def changeset(struct, attrs \\ %{}) do
+    struct
+    |> cast(attrs, [
+      :badge,
+      :type,
+      :rail_line,
+      :expires
+    ])
+    |> unique_constraint([:badge, :type])
+    |> validate_required([:expires])
```

Comment:
why aren't they all required?

Does `unique_constraint` imply that the fields exist?

Reply:
Oh, I thought it did—but I looked, and no. fixed! https://github.com/mbta/orbit/pull/203/files#diff-10334beccc32492f2c1511429be01af770e8fd19351dffbde7df02f4c39b9a55R32